megatron/training.py

```diff
@@ -461,13 +479,40 @@ def setup_model_and_optimizer(model_provider_func, teacher=False):
     if args.deepspeed:
         print_rank_0("DeepSpeed is enabled.")
         pp = mpu.get_pipeline_model_parallel_world_size()
-        model, optimizer, _, lr_scheduler = deepspeed.initialize(
-            model=model[0],
-            optimizer=optimizer,
-            args=args,
-            lr_scheduler=lr_scheduler,
-            mpu=mpu if args.no_pipeline_parallel else None
-        )
+        if args.data_efficiency_curriculum_learning and build_train_valid_test_datasets_provider is not None:
+            train_ds = None
+            if mpu.get_tensor_model_parallel_rank() == 0:
+                # Number of train/valid/test samples.
+                if args.train_samples:
+                    train_samples = args.train_samples
+                else:
+                    train_samples = args.train_iters * args.global_batch_size
+                eval_iters = (args.train_iters // args.eval_interval + 1) * \
```

Comment:
It seems the original training does not require specifying eval_iters. Does this mean the original training and the CL training would receive different evaluation iterations?

Reply:
Added the following comment: eval_iters and test_iters here are not actually used, only for satisfying the input of build_train_valid_test_datasets_provider. We only need to build the training data here. And we follow baseline's logic to build eval/test dataset later in build_train_valid_test_data_iterators.

Reply:
Looks good.